apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/page.tsx

```diff
@@ -13,53 +14,75 @@ import {
 import { Address } from "#/subgraph/src/scripts/last-addr";
 import { getIpfsMetadata } from "@/utils/ipfsUtils";
 import { pointSystems, proposalTypes } from "@/types";
+import useSubgraphQueryByChain from "@/hooks/useSubgraphQueryByChain";
 import { CV_SCALE_PRECISION } from "@/utils/numbers";
+import { useEffect, useMemo, useState } from "react";
+import LoadingSpinner from "@/components/LoadingSpinner";
 
 export const dynamic = "force-dynamic";
 
 export type AlloQuery = getAlloQuery["allos"][number];
 
-const { urqlClient } = initUrqlClient();
-
-export default async function Pool({
+export default function Pool({
   params: { chain, poolId, garden },
 }: {
   params: { chain: string; poolId: number; garden: string };
 }) {
-  const { data } = await queryByChain<getPoolDataQuery>(
-    urqlClient,
+  const { data, fetching } = useSubgraphQueryByChain<getPoolDataQuery>(
     chain,
     getPoolDataDocument,
     { poolId: poolId, garden: garden },
+    {},
+    {
+      topic: "pool",
+      id: poolId,
+      chainId: chain,
+    },
   );
-  const strategyObj = data?.cvstrategies?.[0] as CVStrategy;
+  const strategyObj = data?.cvstrategies?.[0] as CVStrategy | undefined;
   //const { tooltipMessage, isConnected, missmatchUrl } = useDisableButtons();
 
-  if (!strategyObj) {
-    return <div>{`Pool ${poolId} not found`}</div>;
-  }
+  const maxRatioDivPrecision =
+    (Number(strategyObj?.config?.maxRatio || 0) / CV_SCALE_PRECISION) * 100;
+  data?.cvstrategies?.[0] as CVStrategy;
 
   const pointSystem = data?.cvstrategies?.[0].config?.pointSystem;
-  const strategyAddr = strategyObj.id as Address;
-  const communityAddress = strategyObj.registryCommunity.id as Address;
+  const strategyAddr = strategyObj?.id as Address;
+  const communityAddress = strategyObj?.registryCommunity.id as Address;
   const alloInfo = data?.allos[0] as Allo;
   const proposalType = strategyObj?.config?.proposalType as number;
   const poolAmount = strategyObj?.poolAmount as number;
   const tokenGarden = data?.tokenGarden as TokenGarden;
   const metadata = data?.cvstrategies?.[0]?.metadata as string;
   const isEnabled = data?.cvstrategies?.[0]?.isEnabled as boolean;
-  const { title, description } = await getIpfsMetadata(metadata);
 
-  const spendingLimitPct =
-    (Number(strategyObj?.config?.maxRatio) / CV_SCALE_PRECISION) * 100;
+  const [ipfsResult, setIpfsResult] =
+    useState<Awaited<ReturnType<typeof getIpfsMetadata>>>();
 
-  console.log(
-    "maxRatio: " + strategyObj?.config?.maxRatio,
-    "minThresholdPoints: " + strategyObj?.config?.minThresholdPoints,
-    "poolAmount: " + poolAmount,
-  );
+  useEffect(() => {
+    if (metadata) {
+      getIpfsMetadata(metadata).then((data) => {
+        setIpfsResult(data);
+      });
+    }
+  }, [metadata]);
```

Comment:
should we have some loading while it dont finish?

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/62f3fa611d9e6d3642dc52e515e8c95100005af3